datafusion/core/src/dataframe.rs

```diff
@@ -361,6 +364,35 @@ impl DataFrame {
         Ok(DataFrame::new(self.session_state, plan))
     }
 
+    /// Run a count aggregate on the DataFrame and execute the DataFrame to collect this
+    /// count and return it as a usize.
+    /// ```
+    /// # use datafusion::prelude::*;
+    /// # use datafusion::error::Result;
+    /// # #[tokio::main]
+    /// # async fn main() -> Result<()> {
+    /// let ctx = SessionContext::new();
+    /// let df = ctx.read_csv("tests/data/example.csv", CsvReadOptions::new()).await?;
+    /// let length = df.len().await?;
+    /// # Ok(())
+    /// # }
+    /// ```
+    pub async fn len(self) -> Result<usize> {
```

Comment:
Hmm, could it be `count`? For DataFrame-like API, there is commonly `count` for this.

Reply:
yeah that makes sense, changed